src/payment/strategies/adyenv2/adyenv2-payment-strategy.ts

```diff
@@ -242,131 +193,79 @@ export default class AdyenV2PaymentStrategy implements PaymentStrategy {
             }
 
-            additionalActionComponent.mount(`#${containerId}`);
+            additionalActionComponent.mount(`#${containerId || threeDS2ContainerId}`);
 
             if (onLoad) {
                 onLoad(() => {
+                    reject(new PaymentMethodCancelledError());
                     additionalActionComponent.unmount();
                 });
             }
         });
     }
 
-    private _handle3DS2Fingerprint(resultObject: ThreeDS2Result, paymentMethodId: string): Promise<Payment> {
-        return new Promise((resolve, reject) => {
-            if (!this._adyenCheckout) {
-                throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
-            }
-
-            const { onBeforeLoad, onComplete, onLoad } = this._getAdyenV2PaymentInitializeOptions().threeDS2Options;
-
-            const fingerprintComponent = this._adyenCheckout
-                .create(AdyenComponentType.ThreeDS2DeviceFingerprint, {
-                    fingerprintToken: resultObject.token,
-                    onComplete: (fingerprintData: ThreeDS2OnComplete) => {
-                        const fingerprintPaymentPayload = {
-                            ...fingerprintData.data,
-                            paymentData: resultObject.payment_data,
-                        };
-
-                        const paymentPayload = {
-                            methodId: paymentMethodId,
-                            paymentData: {
-                                nonce: JSON.stringify(fingerprintPaymentPayload),
-                            },
-                        };
-
-                        if (onComplete) {
-                            onComplete();
-                        }
+    private _processPayment(error: any, shouldSaveInstrument?: boolean): Promise<any> {
+        if (!(error instanceof RequestError) || !some(error.body.errors, {code: 'additional_action_required'})) {
+            return Promise.reject(error);
+        }
 
-                        resolve(paymentPayload);
+        return this._handleFromAction(error.body.provider_data)
+            .then(payment =>
+                this._store.dispatch(this._paymentActionCreator.submitPayment({
+                    ...payment,
+                    paymentData: {
+                        ...payment.paymentData,
+                        shouldSaveInstrument,
                     },
-                    onError: (error: AdyenError) => reject(error),
-                });
-
-            const { threeDS2ContainerId, additionalActionOptions } = this._getAdyenV2PaymentInitializeOptions();
-
-            if (onBeforeLoad) {
-                onBeforeLoad();
-            }
+                }))
+            );
+    }
 
-            fingerprintComponent.mount(`#${additionalActionOptions.containerId || threeDS2ContainerId}`);
+    private _shouldMountPaymentComponent(paymentMethodName: string): boolean {
+        const excludedPaymentMethods = [
```

Comment:
What's the relationship between `_shouldMountPaymentComponent` and `excludedPaymentMethods` ? should the list be called something like `unmountablePaymentMethods` ?

Reply:
Purpose for that list is to _not_ mount components of those payment method types.